activerecord/lib/active_record/encryption/extended_deterministic_queries.rb

```diff
@@ -44,7 +44,7 @@ def process_arguments(owner, args, check_for_additional_values)
             return args if owner.deterministic_encrypted_attributes&.empty?
 
             if args.is_a?(Array) && (options = args.first).is_a?(Hash)
-              options = options.dup
+              options = options.with_indifferent_access
```

Comment:
I think it would be preferable to use `stringify_keys` here because `with_indifferent_access` is recursive.

Generally speaking in Active Record column names are store as strings, so it's best to cast symbols to strings at the edge of the API.

It might also make sense to store `deterministic_encrypted_attributes` as a list of strings like we do for `ignored_columns` for instance.

Reply:
Okay, thanks for the review. Went with `stringify_keys` instead.

Since `deterministic_encrypted_attributes` is a public API, and that would be a breaking change, I'll hold off on that for now since it will cover a lot of surface area. That cool? Let me know if you'd like me to tackle it either here or in a follow up PR.

Reply:
> Since `deterministic_encrypted_attributes` is a public API

Ah right, well that's unfortunate :/